geo/src/algorithm/affine_ops.rs

```diff
@@ -0,0 +1,272 @@
+use crate::{CoordFloat, CoordNum, Coordinate, Point};
+use std::fmt;
+
+/// A general affine transformation matrix, and associated operations.
+///
+/// Note that affine ops are **already implemented** on most `geo-types` primitives, using this module.
+///
+/// Note that affine transforms using the same numeric type (e.g. [`CoordFloat`](crate::CoordFloat)) can be **composed**,
+/// and the result can be applied to geometries using e.g. [`MapCoords`](crate::MapCoords)
+///
+/// We conceive of `AffineTransform` as a row-major matrix.
+/// 2D affine transforms require six matrix parameters:
+///
+/// `[a, b, d, e, xoff, yoff]`
+///
+/// these map onto the `AffineTransform` rows as follows:
+///
+/// `[[a, b, xoff],`
+///
+/// `[d, e, yoff],`
+///
+/// `[0, 0, 1]]`
+///
+/// The equations for transforming coordinates `(x, y) -> (x', y')` are given as follows:
+///
+/// `x' = ax + by + xoff`
+///
+/// `y' = dx + ey + yoff`
+///
+/// # Examples
+///
+/// ```
+/// use geo::{MapCoords, AffineTransform};
+/// use geo::{line_string, BoundingRect, point, LineString, Centroid};
+/// use approx::assert_relative_eq;
+///
+/// let ls: LineString = line_string![
+///     (x: 0.0f64, y: 0.0f64),
+///     (x: 0.0f64, y: 10.0f64),
+/// ];
+/// let origin = ls.bounding_rect().unwrap().centroid();
```

Comment:
```suggestion
/// let origin = ls.bounding_rect().unwrap().center();
```
(rects have a built in center method, which will be the same as centroid, but we can omit the noise of the Centroid trait in the example)

Reply:
Fixed! But now realise that `center` returns a `Coordinate` and `centroid` returns a `Point`. So displeasing!